service/collector.go

```diff
@@ -198,12 +199,18 @@ func (col *Collector) runAndWaitForShutdownEvent() {
 // setupConfigurationComponents loads the config and starts the components. If all the steps succeeds it
 // sets the col.service with the service currently running.
 func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
-	cp, err := col.set.ParserProvider.Get(ctx)
+	cfgMap := configparser.NewConfigMap()
+
+	ret, err := col.set.ParserProvider.Retrieve(ctx)
 	if err != nil {
 		return fmt.Errorf("cannot load configuration's parser: %w", err)
 	}
 
-	cfg, err := col.set.ConfigUnmarshaler.Unmarshal(cp, col.set.Factories)
+	if err = ret.MergeTo(cfgMap, ""); err != nil {
```

Comment:
Why not a method to get the config instead and let the caller handle the merge with the methods already available on the config map? Any case that it may actually process the config map instead of simply merging into it?

Reply:
Interesting, the reason is that I think this interface will make ConfigSource == Provider so no need for both. That is the reason I did this in a bit more complicated way.

Reply:
Changed to simplified version as suggested, only Get for the moment.